### PR TITLE
[MM-32831] Change show-more background to be highlighted on permanlink views

### PR DESCRIPTION
**Pull request by Willyfrog**
#### Summary

Make background of `Show more` button from big posts to be highlighted as well as the rest of the post

#### Ticket Link
[MM-32831](https://mattermost.atlassian.net/browse/MM-32831)

#### Screenshots
Screenshot 2021-03-04 at 11 42 05

### CODE REVIEW

**Review by nevyangelova — APPROVED** [review]
LGTM, just curious, weren't we migrating from using changeCss, and if so, isn't it a good opportunity to address the highlighted posts using variables?

**Comment by Willyfrog** [comment]
> LGTM, just curious, weren't we migrating from using changeCss, and if so, isn't it a good opportunity to address the highlighted posts using variables?

Ah! I totally agree with you, I was just considering the fix needed for making it work and didn't take into account the migration. I just switched to using variables.

File: sass/components/_post.scss [hunk 1]
```diff
@@ -2006,6 +2006,30 @@
     animation-fill-mode: forwards;
 }
 
+// Handle fade out effect for permalink highlightning
+.app__body .post-list__table .post:not(.current--user).post--highlight .post-collapse__gradient,
+.app__body .post-list__table .post.post--compact.post--highlight .post-collapse__gradient {
+    background:linear-gradient(rgba(var(--mention-highlight-bg-mixed-rgb), 0), rgba(var(--mention-highlight-bg-mixed-rgb), 1));
+}
+
+.app__body .post-list__table .post:not(current--user).post--highlight:not(.post--compact) .post-collapse>.post-collapse__show-more
```
Comment (hmhealey) [inline sass/components/_post.scss:2015]: Is the `.post-collapse>` part of the rule necessary? There's definitely other parts that aren't necessary, but that will keep these consistent with the other rules that look at the post and then the show-more.

File: sass/components/_post.scss [hunk 2]
```diff
@@ -2006,6 +2006,30 @@
...
+{
+    background: rgba(var(--mention-highlight-bg-mixed-rgb), 1);
+}
+
+.app__body .post-list__table .post.current--user.post--highlight:not(.post--compact) .post-collapse__gradient {
+    background:linear-gradient(rgba(var(--own-highlight-bg-rgb), 0), var(--own-highlight-bg-rgb));
+}
+
+.app__body .post-list__table .post.current--user.post--highlight:not(.post--compact) .post-collapse>.post-collapse__show-more {
```
Comment (hmhealey) [inline sass/components/_post.scss:2024]: Same here

File: sass/components/_post.scss (hunk 1 above)
Comment (Willyfrog) [reply sass/components/_post.scss:2015]: Most likely no, but I'll re-check.

**Review by nevyangelova — APPROVED** [review]

**Review by hmhealey — APPROVED** [review]

**Comment by jgilliam17** [comment]
@Willyfrog few small issue
1) The background fade-out is out of sync. See video.

https://user-images.githubusercontent.com/52937121/110807201-cb048e00-8250-11eb-843f-26c3b595b185.mov

2) When long post is pinned, pinned highlight of the "Show more" area is missing
Screen Shot 2021-03-11 at 10 12 21 AM

**Comment by Willyfrog** [comment]
thanks @jgilliam17 ! let me know if now looks better :)

**Comment by jgilliam17** [comment]
@Willyfrog can you adjust permalink animation to fade out more gradually? It seems to be removed suddenly.

**Review by nevyangelova — APPROVED** [review]

**Review by jgilliam17 — APPROVED** [review]
Thanks @Willyfrog 
Tested, looks good to merge.
- Verified `show more` on long posts has the correct highlighted background on permalink and when the long post is pinned.
Also, thank you for adding an issue for follow up to correct the animation https://mattermost.atlassian.net/browse/MM-33895

**Pull request merged**